src/Greenshot.Editor/Drawing/EmojiRenderer.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Windows.Media.Imaging;
+using SixLabors.Fonts;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Drawing.Processing;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using Font = SixLabors.Fonts.Font;
+using FontFamily = SixLabors.Fonts.FontFamily;
+using FontStyle = SixLabors.Fonts.FontStyle;
+using Image = System.Drawing.Image;
+using TextOptions = SixLabors.Fonts.TextOptions;
+
+namespace Greenshot.Editor.Drawing
+{
+    internal static class EmojiRenderer
+    {
+        private static Lazy<FontFamily> _fontFamily = new Lazy<FontFamily>(() =>
+        {
+            using var stream = Assembly.GetCallingAssembly().GetManifestResourceStream("Greenshot.Editor.Resources.TwemojiMozilla.ttf");
+            var fontCollection = new FontCollection();
+            fontCollection.Add(stream);
+            fontCollection.TryGet("Twemoji Mozilla", out var fontFamily);
+            return fontFamily;
+        });
+
+        private static ConcurrentDictionary<string, BitmapSource> _iconCache = new ConcurrentDictionary<string, BitmapSource>();
+
+        public static Image<Rgba32> GetImage(string emoji, int iconSize)
+        {
+            var font = _fontFamily.Value.CreateFont(iconSize, FontStyle.Regular);
+
+            var image = new Image<Rgba32>(iconSize, iconSize);
+
+            RenderEmoji(emoji, font, image);
+            return image;
+        }
+
+        private static void RenderEmoji(string emoji, Font font, Image<Rgba32> image)
+        {
+            var verticalOffset = font.Size * 0.045f;
```

Comment:
I'd be very surprised if this works across all font sizes. I'm curious to see what the issue you are having, we should be correctly calculating any ascenders with the latest code. Left/Top alignment should be consistent also.

Reply:
What I want to achieve is pretty simple : I want to render the emoji perfectly centered in a square bitmap. But it's not what you usually do with a font.

I use horizontal / vertical centering to achieve that but it's slightly offset on the vertical axis. Actually my hack is working well on all sizes, when you resize the emoji in editor it's perfectly smooth.

I don't know if it's how the font is made or something wrong in SixLabor.Fonts.